### ISSUE TITLE
whitescreening on disp-quote elements when contained within footnotes

### ISSUE THREAD

**Issue by axfelix** [substance/texture#256]
Hi guys,

At first I thought we might be producing some invalid JATS, but this should actually be OK (which explains why this is getting past your validator but still crashing Texture).

Basically, it looks like Word for Mac (and only Word for Mac) occasionally formats footnotes such that when they're run through our parsing engine, the resulting JATS comes out like:

`<fn><p><disp-quote><p>Content</p></disp-quote></p></fn>`

This is pretty ugly but actually *should* be OK as far as I can tell. Let me know if you need a doc to reproduce with (not sure if I can share @jalperin's manuscript).

**Comment by michael** [comment]
Hi Alex! Can you provide a minimal test file, containing nothing but the content that causes the error? Just so we can reproduce, then we can address it.

**Comment by axfelix** [comment]
[28-99Z_Article Text-93-1-2-20170906.zip](https://github.com/substance/texture/files/1281764/28-99Z_Article.Text-93-1-2-20170906.zip)

Sure, this should do it

**Comment by michael** [comment]
This comes from a normalization step we have taken. We do a transformation to pull out block elements when wrapped inside a <p> element. We do this to pull out fig elements from paragraphs and move them to the top level for instance.

In your footnote case it turn this:

```xml
<fn id="bibd2e184">
  <p>
    <disp-quote>
      <p>foo</p>
    </disp-quote>
  </p>
</fn>
```

into this:

```xml
<fn id="bibd2e184">
  <disp-quote>
    <p>foo</p>
  </disp-quote>
</fn>
```

And this then fails the JATS4R validation, as disp-quote elements are not allowed as childs of fn. It would also not be valid in JATS. Generally I think this paragraph trick opens a backdoor for a lot of things which then are valid in JATS but not intended.

Do we really want to allow disp-quotes in footnotes.. or would you rather tweak your parsing engine to output just a paragraph instead which would be more footnote-ish?

**Comment by axfelix** [comment]
Hey Michael,

We can look at gradually moving away from this behaviour, but for now there are lots of moving parts involved in doing so unless we just write a sanity check at the end of our process, and while I can do that, I'd rather Texture not be whitescreening on technically valid JATS :)

**Comment by oliver----** [comment]
I agree that it would be great if Texture could deal with all the technically possible variants of content. However, this is technically not feasible. We find restrictions which still give us enough expressivity.
Regarding `<p> > <disp-quote>`: we restricted the options to nest block-level content in paragraphs and added a converter which is now creating invalid JATS -- which is obviously not good ;)
We should discuss this in general, i.e. which content is necessary for `<fn>`s. (There are some other similar elements with only `<p>`s)
In the short term, we should fix this specific converter.

**Comment by jalperin** [comment]
It's frustrating to see failed validations. Could we strip out failing
elements and display the stripped out text?

**Comment by michael** [comment]
@Integral can you implement a transformer `j2r/NormalizeFn.js` that recognises this structure:

`<fn><p><disp-quote><p>Content</p></disp-quote></p></fn>`

and turns it into:

`<fn><p>Content</p></fn>`

Please make sure this is run before `UnwrapBlockElements`.

- Maybe you can write it so that it also recognises other structures. After transform it should always only have p elements inside `<fn>`.
- Please also define a strict schema for `<fn>` in JATS4R to only allow p elements.
- Add a test for it

Thanks a lot!

**Comment by Integral** [comment]
@michael pls review #269

**Comment by michael** [comment]
Fixed via #269
